likecoin/public/likecoin-button.php

```diff
@@ -99,18 +85,15 @@ function likecoin_content_filter( $content ) {
 		// default show LikeButton if id exist.
 	} while ( false );
 
-	if ( is_singular( $post_type_query ) && in_the_loop() && is_main_query() ) {
-		if ( ! empty( $post->post_password ) ) {
-			return $content;
-		}
-		return $content . likecoin_add_likebutton();
+	if ( ! is_singular( $post_type_query ) ) {
+		$likecoin_button_widget = '';
```

Comment:
Instead of setting `likecoin_button_widget` with value, then unsetting it if condition not match, can we reorder the code such that it would only be set if all conditions are met?

Reply:
after checking the conditions, the decision point on whether to show the button or not is actually on this part of the code "'none' !== $widget_position" ...the is_singular() has no impact on it. Hence, maybe we can delete is_singular() since it's ===1 for both "show button" & "not show button" posts. However, I will post a version which is_singular() is still present, just to keep the code structure as the same as the history for now.